packages/contracts-bedrock/test/libraries/Blueprint.t.sol

```diff
@@ -4,14 +4,100 @@ pragma solidity 0.8.15;
 import { Test } from "forge-std/Test.sol";
 import { Blueprint } from "src/libraries/Blueprint.sol";
 
+// Used to test that constructor args are appended properly when deploying from a blueprint.
+contract ConstructorArgMock {
+    uint256 public x;
+    bytes public y;
+
+    constructor(uint256 _x, bytes memory _y) {
+        x = _x;
+        y = _y;
+    }
+}
+
+// Foundry cheatcodes operate on the next call, and since all library methods are internal we would
+// just JUMP to them if called directly in the test. Therefore we wrap the library in a contract.
```

Comment:
I understand why you're wrapping this library in a contract so that it's callable from the tests. But I'm not clear on what this means exactly:

```
and since all library methods are internal we would just JUMP to them if called directly in the test. Therefore we wrap the library in a contract.
```

What do you mean by "JUMP to them"?

Reply:
This is referring to the JUMP opcode vs. the CALL opcode. When functions are internal to a contract, they are part of the contract's bytecode, so the solidity compiler either:
- Directly inlines the function logic at the point the function is used, or
- JUMPs to the function 

Usually it's JUMP, but I think sometimes it can be inlined depending on compiler settings, either way this is an implementation detail and that part that matters is that there is no CALL.

This is true of `internal` library functions also, they behave as if they're part of the contract.

Foundry cheatcodes like expectRevert operate on the next CALL only, so we have to make sure we call out to the library methods instead of jumping to them. (The cheats technically can operate on JUMPS if you only have a single expectRevert call, but that was introduced on accident and now people rely on it, but it doesn't work properly and is a huge footgun, so do not use the cheats with JUMPs).

Reply:
Great explainer thanks.